test/C/src/federated/transient/TransientDownstreamWithTimer.lf

```diff
@@ -0,0 +1,158 @@
+/**
+ * This LF program tests if a transient federate corretly leaves then joins the federation. It also
+ * tests if the transient's downstream executes as expected, that is it received correct TAGs,
+ * regardless of the transient being absent or present. In this test:
+ * - the transient federate spontaneously leaves the federation after 2 reactions to input port
+ * `in`,
+ * - the downstream of the transient federate has only one transient as upstream.
+ */
+target C {
+  timeout: 2 s
+}
+
+preamble {=
+  #include <stdlib.h>
+  #include <string.h>
+=}
+
+/** Persistent federate that is responsible for lauching the transient federate */
+reactor TransientExec(offset: time = 0, period: time = 0, fed_instance_name: char* = "instance") {
```

Comment:
Perhaps change `offset` to `launch_time` and remove the `period` parameter, unless you think this will really be useful to launch the transient multiple times.